pandas/core/series.py

```diff
@@ -3548,6 +3551,13 @@ def sort_values(
         values_to_sort = ensure_key_mapped(self, key)._values if key else self._values
         sorted_index = nargsort(values_to_sort, kind, bool(ascending), na_position)
 
+        if array_equal_fast(
```

Comment:
Is it quicker to skip `array_equal_fast` if we know `values_to_sort` had a NA?

Reply:
This is not definitive, if it's at the we could avoid a copy as well